### PR TITLE
ci(dependabot): Add dependabot.yml for GitHub Actions

### PR DESCRIPTION
**Pull request by rasa**
#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.
- [x] I have added an entry in the CHANGELOG.

### CODE REVIEW

**Comment by niheaven** [comment]
Sorry I'm not familiar with GHA, what's this for? Thx

**Comment by rasa** [comment]
> Sorry I'm not familiar with GHA, what's this for? Thx

@niheaven It will generate PRs to tickle us when a new release of a GitHub action occurs, such as for [potatoqualitee/psmodulecache](https://github.com/ScoopInstaller/Scoop/blob/f93028001fbe5c78cc41f59e3814d2ac8e595724/.github/workflows/ci.yml#L17).

**Comment by rasa** [comment]
And with this, we can change our bleeding edge`actions/checkout@main`'s to `actions/checkout@v3`'s which seems more stable, imo.

**Review by niheaven — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,10 @@
 - **git:** Use Invoke-Git() with direct path to git.exe to prevent spawning shim subprocesses ([#5122](https://github.com/ScoopInstaller/Scoop/issues/5122))
 - **scoop-download:** Output more detailed manifest information ([#5277](https://github.com/ScoopInstaller/Scoop/issues/5277))
 
+### Continuous Integration
+
+- **dependabot** Add dependabot.yml for GitHub Actions ([#5377](https://github.com/ScoopInstaller/Scoop/pull/5377))
```
Comment (niheaven) [inline CHANGELOG.md:20]:
```suggestion
- **dependabot:** Add dependabot.yml for GitHub Actions ([#5377](https://github.com/ScoopInstaller/Scoop/pull/5377))
```

**Review by niheaven — APPROVED** [review]

**Pull request merged**